test/Router.js

```diff
@@ -62,6 +62,8 @@ describe('Router', function(){
   })
 
   it('should not stack overflow with many registered routes', function(done){
+    this.timeout(7500) // extra time for set up & run
```

Comment:
I just run some tests and can confirm that this change solves the issue on my local `aarch64` test system, However run into a different timeout:

      1) app.all()
           should add a router per method:
         Error: Timeout of 2000ms exceeded. For async tests and hooks, ensure "done()" is called; if returning a Promise, ensure it resolves. (/opt/express/test/app.all.js)
          at listOnTimeout (node:internal/timers:557:17)
          at processTimers (node:internal/timers:500:7)

Reply:
Darn. That particular tests doesn't do much at all. It seems like you likely just need to blanket increase the timeout for the environment you're running in. Are you about to run the test suite like the following? `npm test -- --timeout 7500` ?

Reply:
@dougwilson I've run some test on real aarch64 hardware. While the test passes there, that particular step is very close to it's maxmimum:

      Router
        ✔ should return a function with router methods
        ✔ should support .use of other routers
        ✔ should support dynamic routes
        ✔ should handle blank URL
        ✔ should handle missing URL
        ✔ should not stack overflow with many registered routes (1209ms)
        .handle

1209ms for a maximum of 2000ms timeout, so apart from the problem on really low end emulated "hardware" I think your suggestion of setting the timeout to a higher value for that test is very well motivated.

You find the full output at https://paste.opensuse.org/89914697

The test machine was a Cavium ThunderX, an I created a VM there with host CPU passthrough and 2 GB of RAM using an openSUSE Leap JeOS image.

Reply:
Ah, I see, there is a an issue with npm passing arguments around. Any reason you are using `npm run test-ci` ? That is meant just for the usage of the GitHub Actions CI here. Why not just use `npm test` ?

Reply:
No particular reasons, `npm test` should work just fine.

Reply:
Ok, I've run some more performance tests using your suggestion. It looks fine here - 5 test runs, 100% passing on the aarch64 VM. The fix is still required, as the affected test requires more then `2000ms`:

        ✔ should not stack overflow with many registered routes (2459ms)
        ✔ should not stack overflow with many registered routes (2312ms)
        ✔ should not stack overflow with many registered routes (2275ms)
        ✔ should not stack overflow with many registered routes (2435ms)
        ✔ should not stack overflow with many registered routes (2265ms)

I adapt your suggestion in this PR and would mark this thread as resolved?